### DISCUSSION TITLE
Loco position system

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#1023] [Q&A]
Hi, I am doing a research want to use UWB system to do indoor navigation in the greenhouse. Because the greenhouse have some plants.  My professor advise me put the all anchors in the ceil to avoid NLOS. So at the begining, the crazyflie can follow my set point to fly but not accuracy(this is not important now for me), but when I run second time, it will crash. Just like roll the dice, sometimes good sometimes bad. Can anyone help me where I need to improve ? Thanks. 
I provide my code and video give your reference. 
![05AFEB64-2BE0-46A1-9DC9-67855E17B8AC_1_102_a](https://github.com/bitcraze/discussions/assets/118436188/9aef527d-9dbc-4b07-83c9-0adf47f45346)
<img width="621" alt="Screenshot 2023-11-07 at 14 28 41" src="https://github.com/bitcraze/discussions/assets/118436188/c6053a22-4a43-4137-84b0-7001ceb978a3">

**Comment by CalvinKoung** [comment]
The video is in there. 
https://youtube.com/shorts/BaDl6YeCh54?feature=share

**Comment by krichardsson** [comment]
I don't have a good answer but some questions

Which mode are you running the system in, TWR?

>  but when I run second time, it will crash. Just like roll the dice, sometimes good sometimes bad.

Not sure what is happening, are you resetting the estimator between flights? I don't think it should be necessary but it could be an interesting experiment. 
Since your anchors are all in one plane, the mirrored position above the plane is as plausible as the real position to the estimator and a possible scenario is that the estimator locks to the wrong position. I don't see why this would happen though, if it has converged to the proper position (as it has done in your video) it should stick to this solution also after landing.

Have you logged the estimated position to see if it looks OK after the first flight?

**Reply by CalvinKoung** [reply]
Yes, I use TWR mode. 
I guess maybe the laptop wifi or 5G signal affect the UWB signals ? Because I turn off the wifi and cellphone signals it fly normal.

**Reply by krichardsson** [reply]
Interesting! It is possible that they can interfere, the default [UWB configuration uses channel 2](https://github.com/bitcraze/crazyflie-firmware/blob/dbb9df1137f11d4e7e3771c56d25a7137b5b69cc/src/deck/drivers/src/locodeck.c#L530) which has a center frequency of 3993.6 MHz and bandwidth of 499.2 MHz.
